### PR TITLE
feat: add tarball details for published packages

### PR DESCRIPTION
**Pull request by mbtools**
Let's try this again. Tested with `pnpm run local:publish:release` and it was working great 😀 

This adds a function to get the tarball details that are currently missing when publishing to verdaccio
(see https://github.com/orgs/verdaccio/discussions/2974).

The version manifest will now include `{ dist: { fileCount: ..., unpackedSize: ... } }`.

Before:

image

After:

image

### CODE REVIEW

**Comment by juanpicado** [comment]
Great thanks, :) I'm a bit AKF but I'll check as soon I can

File: packages/store/src/storage.ts [hunk 1]
```diff
@@ -1905,6 +1915,25 @@ class Storage {
...
+      version.dist == undefined ||
+      version.dist?.fileCount == undefined ||
+      version.dist?.unpackedSize == undefined
+    ) {
+      debug('tarball stats not found, calculating');
+      try {
+        return await getTarballDetails(buffer);
+      } catch (err: any) {
+        logger.error({ err: err.message }, 'getting tarball details has failed: @{err}');
+        throw err;
```
Comment (juanpicado) [inline packages/store/src/storage.ts:1930]: 🤔 not very sure about throwing here might have side effects or just loggin the error

**Review by juanpicado — APPROVED** [review]
LGTM, great addition 🙇🏼 , just small comment.

File: packages/store/src/storage.ts (hunk 1 above)
Comment (mbtools) [reply packages/store/src/storage.ts]: I moved the getTarballStats call into an existing try catch. That should be better

**Pull request merged**